### PR TITLE
Add integration relationships tab to contact details' page

### PR DESCRIPTION
**Pull request by virlatinus**
| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

#### Description:
This PR creates a new tab beside Social in the contact details' page to display integration relationships based on values in the integration_entity table.

#### Steps to test this PR:
1. Apply this PR
2. Go to a contact details' page. You'll notice the new Integrations tab beside the Social Tab
3. Click it to view the integration relationships. If the contact doesn't have any, you can create one by pushing the contact to an integration. You can also fetch leads with the `app/console mautic:integration:fetchleads -i INTEGRATION` command.

### CODE REVIEW

File: app/bundles/LeadBundle/Controller/LeadController.php [hunk 1]
```diff
@@ -345,7 +345,9 @@ public function viewAction($objectId)
 
         // We need the EmailRepository to check if a lead is flagged as do not contact
         /** @var \Mautic\EmailBundle\Entity\EmailRepository $emailRepo */
-        $emailRepo = $this->getModel('email')->getRepository();
+        $emailRepo       = $this->getModel('email')->getRepository();
+        $em              = $this->factory->getEntityManager();
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Controller/LeadController.php]: Don't use factory - get directly from container

File: app/bundles/LeadBundle/Translations/en_US/messages.ini [hunk 2]
```diff
@@ -462,6 +463,7 @@ mautic.lead.report.points.delta="Point change"
 mautic.lead.report.points.event_name="Point event name"
 mautic.lead.report.points.table="Contact Point Log"
 mautic.lead.report.points.type="Point event type"
+mautic.lead.integrations.header="No Integration Relationships Found"
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Translations/en_US/messages.ini]: Let's not use camel case. Just "No integration relationships found"

File: app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php [hunk 3]
```diff
@@ -389,6 +389,65 @@ public function findLeadsToCreate($integration, $leadFields, $limit = 25, $fromD
...
+        $q = $this->_em->getConnection()->createQueryBuilder()
+            ->from(MAUTIC_TABLE_PREFIX.'integration_entity', 'i');
+
+        if (false === $limit) {
+            $q->select('count(*) as total');
+        } else {
+            $q->select('i.integration, i.integration_entity, i.internal_entity, i.integration_entity_id, i.date_added, i.last_sync_date, i.internal');
+        }
+
+        $q->where(
```
Comment (alanhartless) [inline app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php]: These where statements need to be ordered to match one of the existing indexes for this table.

**Comment by virlatinus** [comment]
@alanhartless I pushed the changes you requested. Thanks!

**Comment by mqueme** [comment]
This works, however the timeline is not the same as the rest of the tabs. it might be better to have a toggle to view details to be consistent with other tabs.

**Comment by virlatinus** [comment]
@mqueme the card view was approved by @matishaladiwala

**Comment by mqueme** [comment]
👍

**Pull request merged**